ra_aid/tools/expert.py

```
@@ -1,36 +1,160 @@
+
     except Exception as e:
         _model = None
-        console.print(
-            Panel(
-                f"Failed to initialize expert model: {e}",
-                title="Error",
-                border_style="red",
-            )
-        )
-        raise
+        raise ToolExecutionError(f"Expert model initialization failed: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)